src/mqtt.cpp

```diff
@@ -505,6 +507,23 @@ mqtt_publish_schedule() {
   }
 }
 
+bool
+mqtt_publish_config() {
+  if(!config_mqtt_enabled() || !mqttclient.connected() || evse.getEvseState() == OPENEVSE_STATE_STARTING) {
+    return false;
+  }
+  const size_t capacity = JSON_OBJECT_SIZE(128) + 1024;
+  DynamicJsonDocument doc(capacity);
+  config_serialize(doc, true, false, true);
+  mqtt_publish_json(doc, "/config");
+
+  String fulltopic = mqtt_topic + "/config_version";
+  String payload = String(config_version());
+  mqttclient.publish(fulltopic, payload, true);
```

Comment:
You don't need this, `config_version` is evented via event_send when the version number is incremented

Reply:
> You don't need this, `config_version` is evented via event_send when the version number is incremented 

We need it when the first publication comes from the mqtt layer. Version is not upgraded at startup. I initially didn't have it but if we remove it, only the config is pushed not the version,  and it takes a config change coming from the UI to get it published.

Reply:
But with it here it is published twice. It may be better to handle explicitly in the MQTT connection code. I would also assume the same bug exists for all the xxx_version, that being said are these useful for MQTT if you are exposing the resources directly? They are meant so the version number is evented over MQTT or Websocket, and then you would fetch the resource over HTTP. With the config, etc being sent over MQTT directly the version is irrelevant.

Reply:
> But with it here it is published twice. 

Yes I knew that...

> It may be better to handle explicitly in the MQTT connection code.

Thought about that and no because the version number and config might not be published at mqtt connection time if evse is not started. So the initial publish will come from the loop.

>  I would also assume the same bug exists for all the xxx_version, that being said are these useful for MQTT if you are exposing the resources directly? 

Not at all... There is no point in having have these *_version exposed in mqtt (IMO). Only exception is if someone wants to know hen a config change happened...

> They are meant so the version number is evented over MQTT or Websocket, and then you would fetch the resource over HTTP. With the config, etc being sent over MQTT directly the version is irrelevant.

I agree.

So to keep the behaviour backward compatible but still avoid this duplicate publish, what I can do is check the version number is the mqtt publish function. If it is 1, it means it has to be published because this is the initial publish. If it is greater than 1, then there has been a config change, and the event will be responsible to publish.

Reply:
I've just pushed https://github.com/OpenEVSE/ESP32_WiFi_V4.x/pull/634/commits/ad9524659cd410a822f7c67c872acce02db83ee2 as a proposal to remove the duplication